generate_config.py

```diff
@@ -0,0 +1,91 @@
+import json
+import argparse
+import os
+import yaml
+import shutil
+from tableau_utilities.tableau_file.tableau_file import Datasource
+from tableau_utilities.tableau_server.tableau_server import TableauServer
+
+
+def do_args():
+    """ Parse arguments.
+
+    Returns: an argparse.Namespace
+    """
+
+    parser = argparse.ArgumentParser(description='List columns in Tableau datasources')
+    parser.add_argument(
+        '--server',
+        required=True,
+        help='Tableau Server URL. i.e. <server_address> in https://<server_address>.online.tableau.com',
+        default=None
+    )
+    parser.add_argument(
+        '--site',
+        required=True,
+        help='Site name. i.e. <site> in https://<server_address>.online.tableau.com/#/site/<site>',
+        default=None)
+    parser.add_argument('--api_version', help='Tableau API version', default='2.8')
+    parser.add_argument('--user', help='user name')
+    parser.add_argument('--password', help='password')
+    parser.add_argument('--token_secret', help='Personal Access Token Secret')
+    parser.add_argument('--token_name', help='Personal Access Token Name')
+    parser.add_argument('--datasource', help='The name of the datasources to generate a config for')
+    return parser.parse_args()
+
+
+def generate_config(server, datasource_name):
```

Comment:
Can you split this into two functions, such as (pseudo code):
```python
def download_all_datasources(creds, datasources_dir):
    os.makedirs(datasources_dir, exists_ok=True)
    ts = TableauServer(**creds)
    for ds in ts.get_downloads():
        ts.download_datasource(ds.id, download_dirs)

def generate_column_config_from_datasources_directory(datasources_dir):
    all_columns = list()
    for ds_file in os.listdir(datasources_dir):
        ds = Datasource(os.path.join(datasources_dir, ds_file))
        all_columns.extend([c.dict() for c in ds.columns])
    # etc...
```
This way we can create a config by either downloading all of the datasources from Tableau Online, or from datasources we've already downloaded locally.

Reply:
yeah, definitely will adjust this

Reply:
That said, I think we'll want to do this 1 at a time.  I think there are going to be conflicts between naming and definitions.

My plan for generating a master config is:
1. Download config A
2. Download config B
3. Commit Config A into a branch
4. Write code that will try to merge config B into A.  This will generate a PR that can be reviewed and edited.
5. Merge the code in and i have a master config
6. Download config C, run the merge script, generate a new PR etc

so I don't plan to write code that would just parse an entire repo to create a config from many datasources in 1 pass

Reply:
Ahh okay I gotcha. That's a good plan!

Perhaps we can make this column_config generator flexible enough to do both.

I'm imagining we have the `main` code below look something like this:
```python
if __name__ == '__main__':
    datasources_dir = args.datasources_dir
    if args.download_all:
        datasources_dir = download_all_datasources(creds)
    if datasources_dir:
        create_column_config_from_dir(datasources_dir)
    elif args.datasource_path:
        create_column_config_from_datasource(args.datasource_path)
    elif args.datasource_name:
        ts = TableauServer(**creds)
        ds = [d for d in ts.get_datasources() if d.name == args.datasource_name]
        if ds:
            path = ts.download_datasource(ds[0].id)
            create_column_config_from_datasource(path)
```

Reply:
@JustinGrilli  I'm adding this as an issue for the future.  I think it would be pretty complex to build a config from all the datasources at once.  The biggest issue I see is how you deal with conflicts and choose whether you pick A or B.  I'm building out the configs and basically doing a PR for each new datasource I add. Then I can review the changes and manually resolve before I make a decision on which definitions or calculation to move forward with.

Issue added here: https://github.com/hoverinc/tableau-utilities/issues/18